### PR TITLE
Add codespell GitHub Action

### LINKED ISSUE #28: Add typo spelling GitHub actions
I think your project could benefit from existing GitHub actions that check for typos in code, comments and documentation

### PR DESCRIPTION
**Pull request by ccoVeille**
https://github.com/marketplace/actions/codespell-with-annotations

Fixes #28

### CODE REVIEW

**Comment by ccoVeille** [comment]
@Julien-cpsn it's ready

I simulate someone adding a typo if needed

File: .github/codespell-ignore.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+crate
+ratatui
+lient
+ody
```
Comment (Julien-cpsn) [inline .github/codespell-ignore.txt:4]: Can't find this one in the code base

**Comment by Julien-cpsn** [comment]
Great!
I am not quite sure I understand which benefits it would bring. Can you explain some more please?
Like how is the ouput? Does it make suggestions or correct things directly?

Thanks in advance

File: .github/codespell-ignore.txt (hunk 1 above)
Comment (ccoVeille) [reply .github/codespell-ignore.txt:4]: Patterns like this in readme

```
**B**ody
```

And this in code

```
{B}ody
(B)ody
```

You can install codespell locally and launch it, after removing the lines.

**Comment by ccoVeille** [comment]
> Great!
> I am not quite sure I understand which benefits it would bring. Can you explain some more please?
> Like how is the ouput? Does it make suggestions or correct things directly?
> 
> Thanks in advance

I will create another branch with a typo, to show you.

**Comment by Julien-cpsn** [comment]
@ccoVeille hmmmm I don't see anything special? Maybe I am missing something?

**Comment by ccoVeille** [comment]
@Julien-cpsn The results are way better with typos #38 than #36 

I will close #34 in favor of #37

**Pull request closed without merging**